core/src/main/scala/org/apache/spark/api/python/PythonRDD.scala

```diff
@@ -146,7 +145,10 @@ private[spark] class PythonRDD(
                 accumulator += Collections.singletonList(update)
               }
               if (stream.readInt() == SpecialLengths.END_OF_STREAM) {
```

Comment:
Similarly, it might be nice to add a comment here to clarify that `END_OF_STREAM` is used for detecting whether workers are in a re-usable state.